AsyncDisplayKitTests/ASDisplayNodeImplicitHierarchyTests.m

```diff
@@ -109,8 +102,9 @@ - (void)testCalculatedLayoutHierarchyTransitions
   ASDisplayNode *node3 = [[ASDisplayNode alloc] init];
   
   ASSpecTestDisplayNode *node = [[ASSpecTestDisplayNode alloc] init];
-  node.layoutSpecBlock = ^(ASSizeRange constrainedSize, NSNumber *layoutState){
-    if ([layoutState isEqualToNumber:@1]) {
+  __weak ASSpecTestDisplayNode *weakNode = node;
+  node.layoutSpecBlock = ^(ASSizeRange constrainedSize){
+    if ([weakNode.layoutState isEqualToNumber:@1]) {
```

Comment:
Hmm, this raises an interesting question.  Should the block take an argument of the node it is laying out?  Because self is not always available like it normally is in layoutSpecThatFits, and there is a substantial risk of retain cycles if a developer ever makes a small mistake and uses an instance variable inside one...

Maybe the header documentation should include a warning about the risk of retain cycles...